internal/controllers/language_detect/language_detect.go

```diff
@@ -235,3 +237,58 @@ func (ld *LanguageDetect) isBatFileOrShellFile(lang string) bool {
 	return strings.EqualFold(lang, "Batchfile") ||
 		strings.EqualFold(lang, "Shell")
 }
+
+// This method was added because if o project analyzed is submodule is not possible run tool gitleaks,
+// then was added this logic for check if .git in cloned folder in .horusec is a file.
+// Case yes, we can find where is original git folder and replace in .horusec folder for run gitleaks tool
+// without problems.
+// Case is a folder we can go to next step without breaking changes.
```

Comment:
I think that would be better write this comments like a "normal" godoc. 
```go
// copyGitFolderWhenIsSubmodule do this, this and this, because of this, this and this...
```

Reply:
fixed